### PR TITLE
Add XO as a linter/fixer for TypeScript files

### PR DESCRIPTION
**Pull request by chaucerbao**
Replicated from [ale_linters/javascript/xo.vim](https://github.com/w0rp/ale/blob/master/ale_linters/javascript/xo.vim)

### CODE REVIEW

**Comment by chaucerbao** [comment]
Should address #2389

File: ale_linters/typescript/xo.vim [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+function! ale_linters#typescript#xo#GetExecutable(buffer) abort
+    return ale#node#FindExecutable(a:buffer, 'typescript_xo', [
+    \   'node_modules/.bin/xo',
+    \])
+endfunction
+
+function! ale_linters#typescript#xo#GetCommand(buffer) abort
+    return ale#Escape(ale_linters#typescript#xo#GetExecutable(a:buffer))
+    \   . ale#Pad(ale#Var(a:buffer, 'typescript_xo_options'))
+    \   . ale#Pad('--reporter unix --stdin --stdin-filename %s')
```
Comment (w0rp) [inline ale_linters/typescript/xo.vim]:
```suggestion
    \   . ' --reporter unix --stdin --stdin-filename %s'
```

You can change this back. `ale#Pad()` is just useful for adding a space before options if they aren't an empty string.

**Comment by chaucerbao** [comment]
Hold up on this PR, please. It's possible that simply adding `xo` to `ale_fixers` will work.

```
let g:ale_fixers={
	\'typescript': ['xo']
\}
```

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**